test/hotspot/jtreg/runtime/Safepoint/TestAbortVMOnSafepointTimeout.java

```diff
@@ -23,79 +24,57 @@
  */
 
 public class TestAbortVMOnSafepointTimeout {
 
     public static void main(String[] args) throws Exception {
-        if (args.length > 0) {
-            int result = test_loop(3);
-            System.out.println("This message would occur after some time with result " + result);
-            return;
-        }
-
-        testWith(500, 500);
-    }
-
-    static int test_loop(int x) {
-        int sum = 0;
-        if (x != 0) {
-            // Long running loop without safepoint.
-            for (int y = 1; y < Integer.MAX_VALUE; ++y) {
-                if (y % x == 0) ++sum;
-            }
-        }
-        return sum;
-    }
-
-    public static void testWith(int sfpt_interval, int timeout_delay) throws Exception {
-        // -XX:-UseCountedLoopSafepoints - is used to prevent the loop
-        // in test_loop() to poll for safepoints.
-        // -XX:LoopStripMiningIter=0 and -XX:LoopUnrollLimit=0 - are
-        // used to prevent optimizations over the loop in test_loop()
-        // since we actually want it to provoke a safepoint timeout.
-        // -XX:-UseBiasedLocking - is used to prevent biased locking
-        // handshakes from changing the timing of this test.
         ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xbootclasspath/a:.",
                 "-XX:+UnlockDiagnosticVMOptions",
-                "-XX:-UseBiasedLocking",
+                "-XX:+WhiteBoxAPI",
                 "-XX:+SafepointTimeout",
                 "-XX:+SafepointALot",
                 "-XX:+AbortVMOnSafepointTimeout",
-                "-XX:SafepointTimeoutDelay=" + timeout_delay,
-                "-XX:GuaranteedSafepointInterval=" + sfpt_interval,
-                "-XX:-TieredCompilation",
-                "-XX:-UseCountedLoopSafepoints",
-                "-XX:LoopStripMiningIter=0",
-                "-XX:LoopUnrollLimit=0",
-                "-XX:CompileCommand=compileonly,TestAbortVMOnSafepointTimeout::test_loop",
-                "-Xcomp",
+                "-XX:SafepointTimeoutDelay=50",
+                "-XX:GuaranteedSafepointInterval=1",
                 "-XX:-CreateCoredumpOnCrash",
                 "-Xms64m",
-                "TestAbortVMOnSafepointTimeout",
-                "runTestLoop"
+                "TestAbortVMOnSafepointTimeout$Test",
+                "999" /* 999 is max unsafe sleep */
         );
 
         OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldMatch("Timed out while spinning to reach a safepoint.");
         if (Platform.isWindows()) {
             output.shouldMatch("Safepoint sync time longer than");
         } else {
             output.shouldMatch("SIGILL");
             if (Platform.isLinux()) {
                 output.shouldMatch("(sent by kill)");
             }
-            output.shouldMatch("TestAbortVMOnSafepointTimeout.test_loop");
```

Comment:
Can we check for another frame like e.g. WB_WaitUnsafe? AbortVMOnSafepointTimeout is designed to provide a stack trace of the thread which is blocking the safepoint.

Reply:
Since top frame is always different (platform dependent, e.g. clock_nanosleep) I removed that.
But "sun.hotspot.WhiteBox.waitUnsafe" or "TestAbortVMOnSafepointTimeout$Test.main" should be below, we could use something like that.

So sure I'll add it back with "TestAbortVMOnSafepointTimeout$Test.main".

Reply:
I update comment before mail was sent, still the old comment was sent, please see edited comment :)

Reply:
You're rigth, OutputAnalyzer can only see the top frame which is platform dependent. I think scanning the hs_err file would be too complicated. So I'm ok with omitting the check.